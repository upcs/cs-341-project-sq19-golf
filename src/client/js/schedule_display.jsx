import React, { Component } from 'react';
import {Link} from 'react-router-dom';
import { Schedules } from './input_display'
import '../css/styles.css';

//All possible schedules
export class SchedulesContainer extends Component {
  constructor(props) {
    super(props);
    this.state = { schedules: props.schedules || [] };
  }

  render() {
    let schedules = Schedules.viableSchedules || this.state.schedules;

    let schedulesList = [];
    if (schedules) {
      schedulesList = schedules.map((schedule, i) => {
        return (
          <ScheduleDisplay key={"schedule-" + i} schedule={schedule}/>
        )
      });
    }

    //TODO: Needs date/professor information
    return (
      <section id="main">
		<div id="name">Schedule Name:
			<input id="scheduleName" type="text" placeholder="Enter Schedule Name Here"/>
		</div>
        <div className="horiz-container">
          {schedulesList}
        </div>
        <Link to="/">
          <button id="save" type="button">
            Save
          </button>
        </Link>
		<Link to="/availability">
          <button id="return" type="button">
            Return
          </button>
        </Link>
      </section>
    );
  }
}

//A single schedule
export class ScheduleDisplay extends Component {
  constructor(props) {
    super(props);
    this.state = { schedule: props.schedule };
  }

  render() {
    let schedule = this.state.schedule;

    let classDisplayList = schedule.map((classData, i) => {
      return (
        <ClassDisplay key={"class-" + i} classData={classData}/>
      )
    });

    return (
      <span className="scheduleOption">
        {classDisplayList}
      </span>
    )
  }
};

//A single course
export class ClassDisplay extends Component {
  constructor(props) {
    super(props);
    this.state = { classData: props.classData };
  }

  render() {
    let classData = this.state.classData;

    return (
      <div className="scheduleClass">
        <span className="timeLabel">
          {classData.start} - {classData.end}
        </span>
        <span className="classLabel">
          {classData.title}
          {/*<br/>
          {classData.profName} */}
        </span>
      </div>
    );
  }
};
