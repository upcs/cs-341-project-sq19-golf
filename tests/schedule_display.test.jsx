import React from 'react';
import { shallow, render, setState } from 'enzyme';

import {SchedulesContainer, ScheduleDisplay, ClassDisplay} from '../src/client/js/schedule_display.jsx';

//Mocked functions
beforeEach(() => {
});

//Tests
describe('SchedulesContainer', () => {
  test('Should render correctly', () => {
    //Mocks class data state through providing abitrary properties
    let schedules = [[], []];

    const schedulesContainer = shallow(<SchedulesContainer schedules={schedules}/>);
    expect(schedulesContainer).toMatchSnapshot();
  });
});

describe('ScheduleDisplay', () => {
  test('Should render correctly', () => {
    //Mocks class data state through providing abitrary properties
    let schedule = [[], []];

    const scheduleDisplay = shallow(<ScheduleDisplay schedule={schedule}/>);
    expect(scheduleDisplay).toMatchSnapshot();
  });
});

describe('ClassDisplay', () => {
  test('Should render correctly', () => {
    //Mocks class data state through providing abitrary properties
    let classData = {'start': 0, 'end': 1, 'title': 'class'};

    const classDisplay = shallow(<ClassDisplay classData={classData}/>);
    expect(classDisplay).toMatchSnapshot();
  });
});
