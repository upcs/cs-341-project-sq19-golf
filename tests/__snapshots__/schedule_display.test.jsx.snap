// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassDisplay Should render correctly 1`] = `
<div
  className="scheduleClass"
>
  <span
    className="timeLabel"
  >
    0
     - 
    1
  </span>
  <span
    className="classLabel"
  >
    class
  </span>
</div>
`;

exports[`ScheduleDisplay Should render correctly 1`] = `
<span
  className="scheduleOption"
>
  <ClassDisplay
    classData={Array []}
    key="class-0"
  />
  <ClassDisplay
    classData={Array []}
    key="class-1"
  />
</span>
`;

exports[`SchedulesContainer Should render correctly 1`] = `
<section
  id="main"
>
  <div
    id="name"
  >
    Schedule Name:
    <input
      id="scheduleName"
      placeholder="Enter Schedule Name Here"
      type="text"
    />
  </div>
  <div
    className="horiz-container"
  >
    <ScheduleDisplay
      key="schedule-0"
      schedule={Array []}
    />
    <ScheduleDisplay
      key="schedule-1"
      schedule={Array []}
    />
  </div>
  <Link
    replace={false}
    to="/"
  >
    <button
      id="save"
      type="button"
    >
      Save
    </button>
  </Link>
  <Link
    replace={false}
    to="/availability"
  >
    <button
      id="return"
      type="button"
    >
      Return
    </button>
  </Link>
</section>
`;
